ramlfications/utils/types.py

```diff
@@ -82,48 +84,52 @@ def parse_examples(raml_version, data):
     #
     data = data or {}
     kw = {}
-    example = data.get("example", None)
+    example = data.get("example")
 
     if raml_version == "0.8":
         if "examples" in data:
             del data["examples"]
-            # Should emit a lint warning if there's an examples node and
+            # TODO: Emit a lint warning if there's an examples node and
             # we're processing RAML 0.8: authors may be expecting it to
             # be honored.
         kw["example"] = example
 
-    if raml_version >= "1.0":
+    if raml_version != "0.8":
         if "example" in data:
             kw["example"] = parse_example(None, example)
 
+        kw["examples"] = None
         if "examples" in data:
             if "example" in data:
-                raise RuntimeError("example and examples cannot co-exist")
+                # TODO: Emit a lint warning during validation.
+                raise InvalidRAMLStructureError(
+                    "example and examples cannot co-exist")
             examples = data["examples"]
             # Must be a map:
             if not isinstance(examples, dict):
                 # Need to decide what exception to make this.
-                raise UnknownDataTypeError
+                raise InvalidRAMLStructureError("examples must be a map node")
             kw["examples"] = NodeList([parse_example(nm, val)
                                        for nm, val in iteritems(examples)])
-        else:
-            kw["examples"] = None
 
     return kw
 
 
 def parse_example(name, node):
     data = dict(name=name, value=node)
     if name:
-        data["display_name"] = BaseContent(name)
+        data["display_name"] = name
     if isinstance(node, dict):
         # Might have a 'value' key; adds a layer.
         if "value" in node:
-            data = node
-            data["name"] = name
+            data["value"] = node["value"]
             if "description" in node:
-                data["description"] = BaseContent(data["description"])
-            if "displayName" in data:
-                data["display_name"] = BaseContent(data.pop("displayName"))
+                data["description"] = BaseContent(node["description"])
+            if "displayName" in node:
+                data["display_name"] = node["displayName"]
+            if "strict" in node:
+                data["strict"] = node["strict"]
```

Comment:
I think with these things (side-stepping the eventual adding of `NOTHING` or some type of `Undefined` type), I think it'd be cleaner if it was like this:

```python
data["description"] = BaseContent(node.get("description", ""))
data["display_name"] = node.get("displayName", name)
data["strict"] = node.get("strict", True)
```

You'd be able to remove the following 2 lines, too.